utils/build-script

```diff
@@ -353,6 +353,21 @@ class BuildScriptInvocation(object):
             args.skip_test_tvos = True
             args.skip_test_watchos = True
 
+        # --build-swift-stdlib is an alias for --build-swift-dynamic-stdlib=1
+        # and --build-swift-static-stdlib=0, which takes precedence over those
+        # options.
+        if args.build_swift_stdlib:
+            args.build_swift_dynamic_stdlib = True
+            args.build_swift_static_stdlib = False
```

Comment:
You're right.  We probably want it to be a tristate then.  If `--build-swift-stdlib` is not passed on the command line, do nothing, if it is passed, then it should override other args.

Once this PR is merged, I'll try to remove the uses of legacy parameters internally.